packages/quantic/force-app/main/default/lwc/quanticSourceCitations/__tests__/quanticSourceCitations.test.js

```diff
@@ -0,0 +1,196 @@
+      await flushPromises();
+
+      const citations = element.shadowRoot.querySelectorAll(selectors.citation);
+      expect(citations).not.toBeNull();
+      expect(citations.length).toBeGreaterThan(0);
+      expect(citations.length).toEqual(mockCitations.length);
+
+      for (let i = 0; i < citations.length; i++) {
+        expect(citations[i].getAttribute('data-key')).toEqual(mockCitations[i].id);
+      }
+    });
+
+    it('should display the proper citation index', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const citationsIndices = element.shadowRoot.querySelectorAll(
+        selectors.citationIndex
+      );
+
+      expect(citationsIndices).not.toBeNull();
+      expect(citationsIndices.length).toEqual(mockCitations.length);
+
+      for (let i = 0; i < citationsIndices.length; i++) {
+        const indexAsString = (i + 1).toString();
+        expect(citationsIndices[i].textContent).toEqual(indexAsString);
+      }
+    });
+
+    it('should properly display the citation links', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const citationLinks = element.shadowRoot.querySelectorAll('a');
+
+      expect(citationLinks).not.toBeNull();
+      expect(citationLinks.length).toEqual(mockCitations.length);
+
+      for (let i = 0; i < citationLinks.length; i++) {
+        expect(citationLinks[i].getAttribute('title')).toEqual(
+          mockCitations[i].title
+        );
+        expect(citationLinks[i].getAttribute('href')).toEqual(
+          mockCitations[i].clickUri
+        );
+      }
+    });
+
+    describe('when a citation is clicked', () => {
+      it('should properly open the citation in a new tab', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        const citationLinks = element.shadowRoot.querySelectorAll('a');
+
+        expect(citationLinks).not.toBeNull();
+        expect(citationLinks.length).toEqual(mockCitations.length);
+
+        for (let i = 0; i < citationLinks.length; i++) {
+          expect(citationLinks[i].getAttribute('target')).toEqual('_blank');
+        }
+      });
+      it('should execute the citationClickHandler function', async () => {
+        const element = createTestComponent();
+        await flushPromises();
+
+        const citations = element.shadowRoot.querySelectorAll(
+          selectors.citation
+        );
+        expect(citations).not.toBeNull();
+        expect(citations.length).toEqual(mockCitations.length);
+
+        citations[0].click();
+        await flushPromises();
+
+        expect(element.citationClickHandler).toHaveBeenCalled();
+        expect(element.citationClickHandler).toHaveBeenCalledWith(
+          mockCitations[0].id
+        );
+      });
+    });
+  });
+
+  describe('when there are no citations found', () => {
+    it('should not display the quantic-source-citations component', async () => {
```

Comment:
This description is not very accurate since the Quantic Source Citation is always displayed.
```suggestion
    it('should not display citations', async () => {
```